modules/core/src/main/java/org/apache/ignite/internal/manager/Producer.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.manager;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.function.BiPredicate;
+
+/**
+ * Interface which can to produce own event.
+ */
+public abstract class Producer<T extends Event> {
+    /** All listeners. */
+    private ConcurrentHashMap<T, ConcurrentLinkedQueue<BiPredicate<List<Object>, Exception>>> listeners = new ConcurrentHashMap<>();
+
+    /**
+     * Registers an event listener.
+     *
+     * @param evt Event.
+     * @param closure Closure.
+     */
+    public void listen(T evt, BiPredicate<List<Object>, Exception> closure) {
```

Comment:
Could you extend javadoc with behavior of listeners if closure returns false or true?

Reply:
Added additional explanation about listen predicate.